openslides/utils/autoupdate.py

```diff
@@ -58,56 +58,62 @@ def validate_absolute_path(self, root, absolute_path):
 
 class OpenSlidesSockJSConnection(SockJSConnection):
     """
-    Sockjs connections for OpenSlides.
+    SockJS connection for OpenSlides.
     """
     waiters = set()
 
-    def on_open(self, request_info):
-        OpenSlidesSockJSConnection.waiters.add(self)
-        self.request_info = request_info
+    def on_open(self, info):
+        self.waiters.add(self)
+        self.connection_info = info
 
     def on_close(self):
         OpenSlidesSockJSConnection.waiters.remove(self)
 
-    def handle_rest_request(self, response):
+    def forward_rest_response(self, response):
         """
-        Handler that is called when the rest api responds.
+        Sends data to the client of the connection instance.
 
-        Sends the response.body to the client.
+        This method is called after succesful response of AsyncHTTPClient().
+        See send_object().
         """
-        # TODO: update cookies
-        if response.code == 200:
-            self.send(response.body)
-
-    @classmethod
-    def send_updates(cls, data):
-        # TODO: use a bluk send
-        for waiter in cls.waiters:
-            waiter.send(data)
+        name, obj_id = get_name_and_id_from_url(response.request.url)
+        data = {
+            'url': response.request.url,
+            'status_code': response.code,
+            'name': name,
+            'id': obj_id,
+            'data': json.loads(response.body.decode())}
+        self.send(data)
 
     @classmethod
     def send_object(cls, object_url):
         """
-        Send OpenSlides objects to all connected clients.
+        Sends an OpenSlides object to all connected clients (waiters).
 
-        First, receive the object from the OpenSlides ReST API.
+        First, retrieve the object from the OpenSlides REST api using the given
+        object_url.
         """
+        # Join network location with object URL.
+        # TODO: Use host and port as given in the start script
+        wsgi_network_location = settings.OPENSLIDES_WSGI_NETWORK_LOCATION or 'http://localhost:8000'
+        url = ''.join((wsgi_network_location, object_url))
+
+        # Send out internal HTTP request to get data from the REST api.
         for waiter in cls.waiters:
-            # Get the object from the ReST API
-            http_client = AsyncHTTPClient()
+            # Read waiter's former cookies and parse session cookie to new header object.
+            session_cookie = waiter.connection_info.cookies[settings.SESSION_COOKIE_NAME]
             headers = HTTPHeaders()
-            # TODO: read to python Morselcookies and why "set-Cookie" does not work
-            request_cookies = waiter.request_info.cookies.values()
-            cookie_value = ';'.join("%s=%s" % (cookie.key, cookie.value)
-                                    for cookie in request_cookies)
-            headers.parse_line("Cookie: %s" % cookie_value)
-
+            headers.add('Cookie', '%s=%s' % (settings.SESSION_COOKIE_NAME, session_cookie.value))
```

Comment:
I think we have only two cookies at the moment: The session cookie and the csrf cookie. Of these two we need only the session cookie.